packages/react-email/build-preview-server.mjs

```diff
@@ -1,24 +1,49 @@
 import { spawn } from 'node:child_process';
 import fs from 'node:fs';
+import path from 'node:path';
+import os from 'node:os';
+import { fileURLToPath } from 'node:url';
 
-const nextBuildProcess = spawn('next', ['build'], {
-  detached: true,
-  stdio: "inherit"
+// Define __filename and __dirname for ES modules
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+// Determine the correct executable based on the operating system
+const isWindows = os.platform() === 'win32';
+const nextExecutable = path.resolve(
+  __dirname,
+  'node_modules',
+  '.bin',
+  isWindows ? 'next.cmd' : 'next'
+);
+
+// Spawn the 'next build' process
+const nextBuildProcess = spawn(nextExecutable, ['build'], {
```

Comment:
If it fails running with just `next build`, would it work by running `pnpm next dev` instead? I do remember pnpm that allows for running package binaries like this, would say it's worth an attempt as it could make this cleaner.

Reply:
appreciate the suggestion. however, it doesn't function as intended for the build process in this context. think we have to keep the existing command

Reply:
I tested in on a Windows machine and I found that the `shell:true` option fixes it, try

```suggestion
const nextBuildProcess = spawn('pnpm', ['next', 'build'], {
  shell: true,
```

And then you won't need the extra code above

Reply:
> I tested in on a Windows machine and I found that the `shell:true` option fixes it, try
> 
> And then you won't need the extra code above

this is definitely more graceful, however, it runs synchronously, keeping the parent process tied to the child process. I’ve tested several configurations and identified three options that work well:

1. `shell: true` **without** `detached`**:**
```javascript
const nextBuildProcess = spawn('pnpm', ['next', 'build'], {
  shell: true,
  stdio: 'inherit',
});
```
2. `shell: true` **with** `detached: true` and `unref()`**:**
```javascript
const nextBuildProcess = spawn('pnpm', ['next', 'build'], {
  shell: true,
  detached: true,
  stdio: 'inherit',
});
nextBuildProcess.unref();
```
3. `windowsHide: true`**:**
```javascript
const nextBuildProcess = spawn('pnpm', ['next', 'build'], {
  shell: true,
  windowsHide: true,
  stdio: 'inherit',
});
```
shall we go with **Option 1** as it’s straightforward, doesn’t involve process detachment, happy to hear your thoughts before updating the pull request.

Reply:
I don't think it matters that much since it is only a build process for CI and for us to run so we shouldn't have to worry.